cli/tests/pcluster/config/test_validators.py

```diff
@@ -2505,3 +2505,44 @@ def test_ebs_volume_size_snapshot_validator(
     utils.assert_param_validator(
         mocker, config_parser_dict, expected_error=expected_error, capsys=capsys, expected_warning=expected_warning
     )
+
+
+@pytest.mark.parametrize(
+    "cluster_section_dict, ebs_section_dict1, ebs_section_dict2, expected_message",
+    [
+        (
+            {"shared_dir": "shared_directory", "ebs_settings": "vol1"},
+            {"volume_size": 30},
+            {},
+            "'shared_dir' can not be specify in the Cluster section when specifying 'ebs_settings'.",
+        ),
+        (
+            {"shared_dir": "shared_directory", "ebs_settings": "vol1"},
+            {"shared_dir": "shared_directory1"},
+            {},
+            "'shared_dir' can not be specify in the Cluster section when specifying 'ebs_settings'.",
+        ),
+        (
+            {"shared_dir": "shared_directory", "ebs_settings": "vol1, vol2"},
+            {"shared_dir": "shared_directory1", "volume_size": 30},
+            {"shared_dir": "shared_directory2", "volume_size": 30},
+            "'shared_dir' can not be specify in the Cluster section when specifying 'ebs_settings'.",
+        ),
+        (
+            {"ebs_settings": "vol1, vol2"},
+            {"shared_dir": "shared_directory1", "volume_size": 30},
+            {"shared_dir": "shared_directory2", "volume_size": 30},
+            None,
+        ),
```

Comment:
can you add a test case for when shared_dir is in the cluster and EBS settings aren't specified?

Reply:
Good idea, done!

Reply:
What i meant by this was a test case where the cluster_section_dict has shared_dir and nothing else.

Reply:
Probably that doesn't need to block this patch though.

Reply:
Fixed!